Helper/Api/Orders.php

```diff
@@ -556,33 +560,43 @@ public function update(Order $order, string $storeKeeperId): void
             return;
         }
 
-        if ($order->getStatus() != 'closed' && $storeKeeperOrder['status'] != 'canceled') {
-            if (isset($statusMapping[$storeKeeperOrder['status']])) {
-                if ($statusMapping[$storeKeeperOrder['status']]
-                    !== $order->getStatus() && $storeKeeperOrder['status']
-                    !== 'complete') {
-                    $this->updateStoreKeeperOrderStatus($order, $storeKeeperId);
+        try {
+            if ($order->getStatus() != 'closed' && $storeKeeperOrder['status'] != 'canceled') {
+                if (isset($statusMapping[$storeKeeperOrder['status']])) {
+                    if ($statusMapping[$storeKeeperOrder['status']]
+                        !== $order->getStatus() && $storeKeeperOrder['status']
+                        !== 'complete') {
+                        $this->updateStoreKeeperOrderStatus($order, $storeKeeperId);
+                    }
                 }
-            }
 
-            $this->updateStoreKeeperOrder($order, $storeKeeperId);
-            $this->createShipment($order, $storeKeeperId);
-        }
+                $paymentId = $order->getStorekeeperPaymentId();
+                if ($order->getStorekeeperPaymentId() && $order->getStatus() !== 'canceled') {
+                    $this->paymentApiClient->attachPaymentIdsToOrder($order->getStoreId(), $storeKeeperId, [$paymentId]);
```

Comment:
```suggestion

                    $this->paymentApiClient->attachPaymentIdsToOrder($order->getStoreId(), $storeKeeperId, [$paymentId]);
```

Payment should be atached even if order was cancelled.